src/Microsoft.ComponentDetection.Common/Telemetry/TelemetryRelay.cs

```diff
@@ -60,10 +62,10 @@ public void Shutdown()
                 try
                 {
                     // Set a timeout for services that flush synchronously.
-                    AsyncExecution.ExecuteVoidWithTimeoutAsync(
+                    await AsyncExecution.ExecuteVoidWithTimeoutAsync(
                         () => service.Flush(),
```

Comment:
Is there a `flushAsync()` available? And maybe the [`Task.Wait(TimeSpan)`](https://learn.microsoft.com/en-us/dotnet/api/system.threading.tasks.task.wait?view=net-6.0#system-threading-tasks-task-wait(system-timespan)) that was added in .NET 6 is better?

Reply:
We might need to create a followup issue to address `AsyncExecution.ExecuteVoidWithTimeoutAsync` calls.

Reply:
no `flushAsync()` is available here. 

the `AsyncExecution.ExecuteVoidWithTimeoutAsync()` has custom exceptions, may be not a good idea to change that logic here by using `Task.Wait(timespan)`.  We can probably keep the api of `ExecuteVoidWithtimeoutAsync()` same and use `Task.Wait(timespan)` inside of it (as part of the follow up issue) ?

Reply:
done